python/pyspark/ml/image.py

```diff
@@ -28,7 +28,7 @@
 from typing import Any, Dict, List, NoReturn, Optional, cast
 
 import numpy as np
-from distutils.version import LooseVersion
+from packaging.version import Version
```

Comment:
Hm, wait. Is this 3rd party library? Adding a new dep is problematic especially when PySpark is installed via downloading distribution (instead of PyPI) because that doesn't handle dependencies (that's why we have to ship Py4J together at https://github.com/apache/spark/tree/master/python/lib)

Reply:
> Is this 3rd party library?

https://pypi.org/project/packaging/ , yes, it is. And in the standard library, seems not have a way to handle this after 3.12.

> Adding a new dep is problematic

Looks like we might have below way to solve this:
- Maintaince the code like `distutils`/`packaging` version in PySpark, just like [cloudpickle](https://github.com/apache/spark/tree/master/python/pyspark/cloudpickle), or at least a simple version implementations.
- Introduce the packaging as 3rd party lib, add deps in setup and [docs](https://spark.apache.org/docs/latest/api/python/getting_started/install.html#dependencies).

BTW, I just think it isn't an ideal way that we always matainence all 3rd deps in pyspark, have we considered the extra third-party dependent installations as a required/option step in the installation of pyspark (especially via downloading distribution)? such as require users install before pyspark startup or install deps in [`bin/pyspark`](https://github.com/apache/spark/blob/master/bin/pyspark) automatically (
This may require additional network access).

Reply:
So copy and paste the Version module like cloudpickle, then?
And if you are on board with this, I want to suggest that we use the current LooseVersion implementation.